src/content/github/file/main.tsx

```diff
@@ -62,22 +60,61 @@ function init(): Promise<void> {
 }
 
 async function main(): Promise<void> {
-  try {
-    if (!isFileUrl(document.URL)) {
-      print("file not detected at current URL");
-      return;
-    }
+  const urlMetadata = getMetadataFromURL();
+  if (!urlMetadata) {
+    print("file not detected at current URL");
+    return;
+  }
 
-    let metadata: FileMetadata;
-    metadata = await getMetadata(document.URL);
+  globals.coverageButton = createCoverageButton();
 
-    globals.coverageButton = createCoverageButton();
+  process(urlMetadata).catch((e) => {
+    print("unexpected error", e);
+    updateButton("Coverage: ⚠");
+  });
+}
 
-    process(metadata)
-  } catch (e) {
-    Sentry.captureException(e)
-    throw e
+function getMetadataFromURL(): { [key: string]: string } | null {
+  const regexp =
+    /\/(?<owner>.+?)\/(?<repo>.+?)\/blob\/(?<branch>.+?)\/(?<path>.+?)$/;
+  const matches = regexp.exec(window.location.pathname);
+  const groups = matches?.groups;
+  if (!groups) {
+    return null;
+  }
+
+  const branch = groups.branch;
+  const commitMatch = branch.match(/[\da-f]+/);
+
+  // branch could be a commit sha
+  if (
+    commitMatch &&
+    commitMatch[0].length == branch.length &&
+    (groups.branch.length === 40 || branch.length === 7)
+  ) {
+    // branch is actually a commit sha
+    let commit = branch;
+
+    // if it's a short sha, we need to get the full sha
+    if (commit.length === 7) {
+      const commitLink = document.querySelector(
+        `[href^="/${groups.owner}/${groups.repo}/tree/${commit}"]`
+      );
```

Comment:
the long sha can be found inside a few links on this page, that's what this code extracts.